diff --git a/stdutil.c b/stdutil.c
index f9ab55a..6e3db70 100644
--- a/stdutil.c
+++ b/stdutil.c
@@ -190,6 +190,8 @@ void systemDeepSleep (void)
   PWR->CR1 =  (PWR->CR1 & (~PWR_CR1_LPMS)) | PWR_CR1_LPMS_SHUTDOWN;
 #elif defined(STM32H7XX)
   PWR->CPUCR |= PWR_CPUCR_PDDS_D1 | PWR_CPUCR_PDDS_D2 | PWR_CPUCR_PDDS_D3;
+#elif defined(STM32G4XX)
+  // TODO
 #else
 #error neither F3XX, F4XX, F7XX, H7XX, L4XX : should be implemented
 #endif
@@ -236,6 +238,8 @@ void systemDeepSleepFromISR (void)
   PWR->CR1 =  (PWR->CR1 & (~PWR_CR1_LPMS)) | PWR_CR1_LPMS_SHUTDOWN;
 #elif defined(STM32H7XX)
   PWR->CPUCR |= PWR_CPUCR_PDDS_D1 | PWR_CPUCR_PDDS_D2 | PWR_CPUCR_PDDS_D3;
+#elif defined(STM32G4XX)
+  // TODO
 #else
 #error neither F3XX, F4XX, F7XX, H7XX, L4XX : should be implemented
 #endif
@@ -454,7 +458,9 @@ const char* getGpioName (const ioportid_t p)
   _GPIOTEST(GPIOG);
 #endif
 #endif
+#if (!defined STM32G4XX)
   _GPIOTEST(GPIOH);
+#endif
 
   return "Unknown GPIO";
 }
diff --git a/stdutil.h b/stdutil.h
index 3bc1626..5462906 100644
--- a/stdutil.h
+++ b/stdutil.h
@@ -341,6 +341,11 @@ F7
 #define BDMA_SECTION ".ram4"    
 #define AXIDMA_SECTION ".ram0nc"
 #define SDMMC_SECTION AXIDMA_SECTION
+#elif  defined STM32G4XX
+#define STD_SECTION ".ram0" 
+#define FAST_SECTION ".ram4" 
+#define DMA_SECTION ".ram0"    
+#define BCKP_SECTION ".ram5"
 #else
 #error "section defined only for STM32F3, STM32F4, STM32F7, STM32L4 and STM32H7"
 #endif
